src/window.py

```diff
@@ -281,6 +284,23 @@ def _move(self, path, navigating=False):
         else:
             self._open(path)
 
+
+            self._places.add_starred(path)
+
+        self._path = path
+        self.bookmark_button.props.sensitive = self._places.canBookmark(path)
+        self._update_bookmark_icon()
+
+    def _update_bookmark_icon(self):
+        if self._places.is_bookmarked(self._path):
+            self.bookmark_button.props.child.props.icon_name = "bookmark-filled-symbolic"
+        else:
+            self.bookmark_button.props.child.props.icon_name = "bookmark-outline-symbolic"
+
+    def _on_bookmark_button_clicked(self, button):
```

Comment:
Then maybe the button widget  could keep track of the path, and not the window. All much more self contained.